tests/test_integration.py

```diff
@@ -351,25 +372,29 @@ class MockServerRequestHandler(MyTestHandler):
         assert req.command == "POST"
         assert isinstance(cus, stripe.Customer)
 
-    @pytest.mark.anyio
-    async def test_async_raw_request_timeout(self):
+    async def test_async_raw_request_timeout(
+        self, set_global_async_http_client
+    ):
         class MockServerRequestHandler(MyTestHandler):
             def do_request(self, n):
                 time.sleep(0.02)
                 return super().do_request(n)
 
         self.setup_mock_server(MockServerRequestHandler)
         stripe.api_base = "http://localhost:%s" % self.mock_server_port
-        stripe.default_http_client = stripe.new_default_http_client()
-        assert isinstance(stripe.default_http_client, stripe.RequestsClient)
-        stripe.default_http_client._async_fallback_client = (
-            stripe.HTTPXClient()
-        )
         # If we set HTTPX's generic timeout the test is flaky (sometimes it's a ReadTimeout, sometimes its a ConnectTimeout)
         # so we set only the read timeout specifically.
-        stripe.default_http_client._async_fallback_client._timeout = (
-            httpx.Timeout(None, read=0.01)
-        )
+        hc = stripe.default_http_client
+
+        expected_message = ""
+        if isinstance(hc, stripe.HTTPXClient):
```

Comment:
Should we unify timeout errors? Makes for a leaky abstraction when to switch HTTP client customers need to rewrite their exception handling.

Reply:
What do you mean by "unify"? The existing logic is that timeout errors will throw a `stripe.APIConnectionError` but have messages derived from the original exception thrown by the underlying client. There's logic in this test that makes assertions specific to text inside the exception because we are testing timeout behavior and we want to be careful that the underlying exception represents a timeout and not some other sort of failure, but I wouldn't expect users to parse `message` in their own exception handling code.